src/main/java/org/jabref/logic/journals/PredatoryJournalLoader.java

```diff
@@ -0,0 +1,127 @@
+package org.jabref.logic.journals;
+
+import java.io.IOException;
+
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+
+import java.net.URI;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PredatoryJournalLoader {
+    private static class PJSource {
+        final String    URL;
+        final String    ELEMENT_REGEX;
+
+        PJSource(String URL, String ELEMENT_REGEX) {
+            this.URL            = URL;
+            this.ELEMENT_REGEX  = ELEMENT_REGEX;
+        }
+    }
+
+    private static final List<PJSource> PREDATORY_SOURCES = List.of(
+        new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/journals.csv",
+                    null),
+        /*
+        new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/hijacked.csv",
+                    null,
+                    null,
+                    "journal", "journalname", "bookname"),
+        */
+        new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/publishers.csv",
+                    null),
+        new PJSource("https://beallslist.net/",
+                    "<li>.*?</li>"),
+        new PJSource("https://beallslist.net/standalone-journals/",
+                    "<li>.*?</li>"),
+        new PJSource("https://beallslist.net/hijacked-journals/",
+                    "<tr>.*?</tr>")
+    );
+
+    private static final Logger                 LOGGER              = LoggerFactory.getLogger(PredatoryJournalLoader.class);
+    private static HttpClient                   client;
+    private static List<String>                 linkElements;
+    private static PredatoryJournalRepository   repository;
+
+    public PredatoryJournalLoader() {
+        this.client             = HttpClient.newHttpClient();
+        this.linkElements       = new ArrayList<>();
+    }
+
+    public static PredatoryJournalRepository loadRepository() {
+        // Initialize in-memory repository
+        repository = new PredatoryJournalRepository();
+
+        // Update from external sources
+        update();
+
+        return repository;
+    }
+
+    private static void update() {
+        PREDATORY_SOURCES   .forEach(PredatoryJournalLoader::crawl);            // populates linkElements (and predatoryJournals if CSV)
+        linkElements        .forEach(PredatoryJournalLoader::clean);            // adds cleaned HTML to predatoryJournals
+
+        LOGGER.info("UPDATED PREDATORY JOURNAL LIST");
+    }
+
+    private static void crawl(PJSource source) {
+        var uri     = URI.create(source.URL);
```

Comment:
You can use our class UrlDownload for downloading/getting data. That already comes with exception handling etc